internal/pkg/template/templates/workloads/services/backend/cf.yml

```diff
@@ -99,25 +88,30 @@ Resources:
     Type: AWS::ECS::Service
     DependsOn:
       - EnvControllerAction
-      {{- if and .ALBEnabled .HTTPSListener}}
-      - HTTPListenerRuleWithDomain
-      - HTTPSListenerRule
-      {{ else if .ALBEnabled}}
-      - HTTPListenerRule
+      {{- if .ALB }}
+      {{- range $i, $rule := .ALB.Rules }}
+      {{- if $.ALB.IsHTTPS}}
+      - HTTPListenerRuleWithDomain{{ if ne $i 0 }}{{ $i }}{{ end }}
+      - HTTPSListenerRule{{ if ne $i 0 }}{{ $i }}{{ end }}
+      {{- else }}
+      - HTTPListenerRule{{ if ne $i 0 }}{{ $i }}{{ end }}
       {{- end}}
+      {{- end }}
+      {{- end }}
     Properties:
       {{- "\n"}}{{ include "service-base-properties" . | indent 6 }}
       ServiceRegistries: !If [ExposePort, [{RegistryArn: !GetAtt DiscoveryService.Arn, Port: !Ref TargetPort}], !Ref "AWS::NoValue"]
-      {{- if .ALBEnabled}}
+      {{- if .ALB}}
       {{- if .HTTPHealthCheck.GracePeriod }}
       HealthCheckGracePeriodSeconds: {{.HTTPHealthCheck.GracePeriod}}
```

Comment:
I wonder if `HTTPHealthCheck` should have been under the `ALB` struct 🤔? If so, I'm okay if we add it in a separate PR as well - this one is already biggg

Reply:
Ohh good question, the reason why I did not add it under `.ALB` was because if use observe L106 then we use this value on a service level here. I think we need to fix this scenario as a part of another PR. Because if we try to get value of the `GracePeriod ` from `.ALB` then we need to get that value from primary listener rule only because every listener rule can have their own specific value. So We need to make decision for this particular `HealthCheckGracePeriodSeconds` param what value should be assigned. That is why kept this param as is for now.

Reply:
> we need to get that value from primary listener rule only because every listener rule can have their own specific value.

Why do we assume that `GracePeriod` is a property of a listener rule?

Reply:
I think it's okay that we leave if outside of `.ALB`, but for a different reason.

This [field](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html#cfn-ecs-service-healthcheckgraceperiodseconds) is used for services with ELB, which can be an ALB or an NLB.

It's fine that we keep it as is ntil we have a manifest field to set the grace period for NLB - for example ⬇️  - 
```
nlb:
  grace_periods: 1s
```

Reply:
> It's fine that we keep it as is ntil we have a manifest field to set the grace period for NLB 

this is another bug I think I have already discussed with Efe and we can fix that separately.